rhino/src/main/java/org/mozilla/javascript/regexp/NativeRegExp.java

```diff
@@ -2784,6 +3085,18 @@ private static int simpleMatch(
                     index = getIndex(program, pc);
                     pc += INDEX_LEN;
                     if (cpInBounds) {
+                        // First, try to match string literals (v flag feature)
+                        int stringLiteralLen =
+                                stringLiteralMatcher(
+                                        gData.regexp.classList[index], input, cpToMatch);
+                        if (stringLiteralLen > 0) {
```

Comment:
zero-length matches are allowed. In node `/[\q{}]/v.exec('')[0]` returns `''`. it